src/transformStateWithClones.js

```
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const property in action.extraData) {
+          stateClone[property] = action.extraData[property];
+        };
+        break;
+
+      case 'removeProperties':
+        for (const property of action.keysToRemove) {
+          delete stateClone[property];
+        };
+        break;
+
+      case 'clear':
+        for (const property in stateClone) {
+          delete stateClone[property];
+        };
+        break;
+
+      default:
+        // eslint-disable-next-line no-console
+        console.log('Check input value');
```

you could throw an error here instead